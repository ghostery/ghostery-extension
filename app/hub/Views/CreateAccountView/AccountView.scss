/**
 * Create Account View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */
@import './partials/_placeholders';
@import './partials/_colors';

// Create Account View
.account-content {
	font-family: Roboto, "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
	font-size: 14px;
	font-weight: bold;
	color: $tundora;
	.row {
		height: 85px;
		margin: 30px;
	}
	.columns {
		width: 343px;
		height: 100%;
		padding: 0 15px 0 15px;
		&.padded {
			padding: 0 42px 0 42px;
		}
		&.wide {
			width: 427px;
			padding: 0;
		}
	}
	input {
		width: 100%;
		height: 41px;
		border: solid 1px #c8c7c2;
		color: $tundora;
	}
	.account-checkbox-container {
		line-height: 85px;
	}
	.account-checkbox {
		width: 24px;
		height: 24px;
		object-fit: contain;
		margin-right: 10px;
	}
	.account-sign-in {
		@extend %pointer;
		color: $ghosty-blue;
		text-decoration: underline;
		height: 30px;
		font-weght: normal;
		line-height: 30px;
	}
	.account-submit {
		@extend %pointer;
		width: 179px;
		line-height: 38px;
		padding: 0 30px 0 30px;
		font-size: 12px;
		font-weight: 500;
		height: 38px;
		border-radius: 3px;
		background-color: #930194;
		color: $white;
		text-transform: uppercase;
		display: inline-block;
		float: right;
	}
	span.asterisk {
		display: none;
		font-size: 14px;
		margin: 0;
	}
	p.warning {
		margin: 0;
		font-size: 12px;
		line-height: 12px;
		font-weight: 400;
		font-style: italic;
		color: #CC5F5A;
		opacity: 0;
		@include transition(opacity 0.2s, color 0.2s);
	}
	.panel-error {
		span.asterisk {display: inline;}
		p.warning {
			opacity: 1;
		}
		input {
			border-color: #e74055;
			background-color: #FFF6F6;
		}
		label {
			color: #e74055;
		}
	}
}
.create-account-result {
	height:41px;
	width: 580px;
	margin: auto;
	position: fixed;
	top: 80px;
	display: none;
	font-size: 13px;
	font-weight: bold;
	line-height: 41px;
	color: #ffffff;
	border-radius: 4px;
	text-align: center;
	&.success {
		display: inline-block;
		background-color: #1fce6d;
	}
	&.error {
		display: inline-block;
		background-color: #e74055;
	}
}
