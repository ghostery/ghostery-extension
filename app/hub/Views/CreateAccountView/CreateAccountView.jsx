/**
 * Create Account View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import { NavLink } from 'react-router-dom';
import { ToggleCheckbox } from '../../../shared-components';

/**
 * A Functional React component for rendering the Create Account View
 * @return {JSX} JSX for rendering the Create Account View of the Hub app
 * @memberof HubComponents
 */
const CreateAccountView = (props) => {
	const {
		email,
		emailError,
		confirmEmail,
		confirmEmailError,
		firstName,
		lastName,
		password,
		passwordInvalidError,
		passwordLengthError,
		legalConsentChecked,
		legalConsentNotCheckedError,
		handleInputChange,
		handleLegalConsentCheckboxChange,
		handleSubmit,
	} = props;

	const emailInputClassNames = ClassNames('CreateAccountView__inputBox', {
		error: emailError,
	});
	const confirmInputClassNames = ClassNames('CreateAccountView__inputBox', {
		error: confirmEmailError,
	});
	const legalConsentCheckboxInputLabelClassNames = ClassNames('CreateAccountView__inputLabel clickable', {
		error: legalConsentNotCheckedError,
	});
	const passwordInputClassNames = ClassNames('CreateAccountView__inputBox', {
		error: passwordInvalidError || passwordLengthError,
	});

	return (
		<div className="CreateAccountView">
			<div className="CreateAccountView--addPaddingTop row align-center">
				<div className="columns small-10">
					<div className="CreateAccountView__header flex-container align-center-middle">
						<img className="CreateAccountView__headerImage" src="/app/images/hub/account/ghosty-account.svg" />
						<div className="CreateAccountView__headerTitle">
							<h3>
								{t('hub_create_account_header_title')}
							</h3>
						</div>
					</div>
				</div>
			</div>
			<form onSubmit={handleSubmit}>
				<div className="CreateAccountView--addPaddingTop row align-center-middle">
					<div className="columns small-12 medium-5">
						<label htmlFor="create-account-email" className="CreateAccountView__inputLabel">
							{t('email_colon')}
						</label>
						<input
							id="create-account-email"
							className={emailInputClassNames}
							name="email"
							type="text"
							value={email}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
						{emailError && (
							<div className="CreateAccountView__inputError">
								{t('please_enter_a_valid_email')}
							</div>
						)}
					</div>
					<div className="columns small-12 medium-5">
						<label htmlFor="create-account-confirmEmail" className="CreateAccountView__inputLabel">
							{t('confirm_email_colon')}
						</label>
						<input
							id="create-account-confirmEmail"
							className={confirmInputClassNames}
							name="confirmEmail"
							type="text"
							value={confirmEmail}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
						{confirmEmailError && (
							<div className="CreateAccountView__inputError">
								{t('your_email_do_not_match')}
							</div>
						)}
					</div>
				</div>
				<div className="row align-center-middle">
					<div className="columns small-12 medium-5">
						<label htmlFor="create-account-firstName" className="CreateAccountView__inputLabel">
							{t('hub_create_account_label_first_name')}
						</label>
						<input
							id="create-account-firstName"
							className="CreateAccountView__inputBox"
							name="firstName"
							type="text"
							value={firstName}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
					</div>
					<div className="columns small-12 medium-5">
						<label htmlFor="create-account-lastName" className="CreateAccountView__inputLabel">
							{t('hub_create_account_label_email_last_name')}
						</label>
						<input
							id="create-account-lastName"
							className="CreateAccountView__inputBox"
							name="lastName"
							type="text"
							value={lastName}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
					</div>
				</div>
				<div className="row align-center-middle">
					<div className="columns small-12 medium-5">
						<label htmlFor="create-account-password" className="CreateAccountView__inputLabel">
							{t('password_colon')}
						</label>
						<input
							id="create-account-password"
							className={passwordInputClassNames}
							name="password"
							type="password"
							value={password}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
						{passwordInvalidError && (
							<div className="CreateAccountView__inputError">
								{t('hub_create_account_label_password_invalid')}
							</div>
						)}
						{passwordLengthError && (
							<div className="CreateAccountView__inputError">
								{t('hub_create_account_label_password_invalid_length')}
							</div>
						)}
					</div>
					<div className="columns small-12 medium-5">
						<div className="CreateAccountView__checkboxContainer CreateAccountView--marginBottom flex-container">
							<ToggleCheckbox
								checked={legalConsentChecked}
								className="ToggleCheckbox--flush-left"
								onChange={handleLegalConsentCheckboxChange}
							/>
							<span
								className={legalConsentCheckboxInputLabelClassNames}
								onClick={handleLegalConsentCheckboxChange}
								dangerouslySetInnerHTML={{ __html: t('create_account_form_legal_consent_checkbox_label') }}
							/>
						</div>
					</div>
				</div>
				<div className="row align-center">
					<div className="CreateAccountView__linkContainer columns small-12 medium-10">
						<p className="CreateAccountView__link">
							{ t('hub_create_account_already_have_account') }
							&nbsp;
							<NavLink to="/log-in">
								{ t('hub_create_account_link_login') }
							</NavLink>
						</p>
					</div>
				</div>
				<div className="row align-center">
					<div className="CreateAccountView__linkContainer columns small-12 medium-10">
						<p className="CreateAccountView__termsStatement" dangerouslySetInnerHTML={{ __html: t('account_creation_privacy_statement') }} />
					</div>
				</div>
				<div className="CreateAccountView--addPaddingTop row align-center">
					<div className="CreateAccountView__submit columns small-12 medium-10 flex-container flex-dir-row-reverse">
						<button type="submit" className="CreateAccountView__button button success">
							{ t('create_account') }
						</button>
					</div>
				</div>
			</form>
		</div>
	);
};

// PropTypes ensure we pass required props of the correct type
CreateAccountView.propTypes = {
	email: PropTypes.string.isRequired,
	emailError: PropTypes.bool.isRequired,
	confirmEmail: PropTypes.string.isRequired,
	confirmEmailError: PropTypes.bool.isRequired,
	firstName: PropTypes.string.isRequired,
	lastName: PropTypes.string.isRequired,
	legalConsentChecked: PropTypes.bool.isRequired,
	password: PropTypes.string.isRequired,
	passwordInvalidError: PropTypes.bool.isRequired,
	passwordLengthError: PropTypes.bool.isRequired,
	handleInputChange: PropTypes.func.isRequired,
	handleLegalConsentCheckboxChange: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
};

export default CreateAccountView;
