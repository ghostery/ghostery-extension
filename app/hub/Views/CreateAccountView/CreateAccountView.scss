/**
 * Create Account View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

$color-create-account-form-error-red: #e74055;

// Create Account View
.CreateAccountView {
	padding-top: 80px;
	padding-bottom: 40px;
}
.CreateAccountView--addPaddingTop {
	padding-top: 40px;
}
.CreateAccountView--addMarginSide {
	margin-left: 10px;
	margin-right: 10px;
}
.CreateAccountView--addMarginBottom {
	margin-bottom: 20px;
}
.CreateAccountView__headerImage {
	max-width: 180px;
}
.CreateAccountView__headerTitle h3 {
	font-size: 24px;
	line-height: 1.75;
	color: #4a4a4a;
	margin: 0 25px;
}
.CreateAccountView__inputLabel {
	font-size: 14px;
	font-weight: 500;
	line-height: 20px;
	color: #4a4a4a;
}
.CreateAccountView__inputBox {
	font-size: 14;
	line-height: 24px;
	color: #4a4a4a;
	margin-bottom: 35px;

	// Foundation Overrides
	border-radius: 0;
	box-shadow: none;
	border: 1px solid #c8c7c2;
}
.CreateAccountView__inputBox.error {
	margin-bottom: 8px;
	border-color: $color-create-account-form-error-red;
}
.CreateAccountView__inputBox:focus {
	// Foundation Overrides
	box-shadow: none;
	border-color: $color-create-account-form-error-red;
}
.CreateAccountView__inputError {
	font-size: 12;
	line-height: 14px;
	color: $color-create-account-form-error-red;
	margin-bottom: 13px;
}
.CreateAccountView__link {
	font-size: 14px;
	line-height: 30px;
}
.CreateAccountView__termsStatement {
	font-size: 12px;
}
.CreateAccountView__button {
	min-width: 180px;
}
.CreateAccountView__inputLabel {
	flex-grow: 1; // ensure left-justify alignment in single column layout
	padding-top: 10px; // align with checkbox top
}
.CreateAccountView__inputLabel.error{
	color: $color-create-account-form-error-red;
}
@media only screen and (max-width: 740px)  {
	.CreateAccountView {
		padding-top: 20px;
	}
	.CreateAccountView__header {
		flex-direction: column;
	}
	.CreateAccountView__headerTitle {
		padding-top: 30px;
		text-align: center;
	}
	.CreateAccountView__submit {
		display: flex;
		justify-content: center;
	}
	.CreateAccountView__checkboxContainer {
		padding-right: 12px;
		justify-content: center;
	}
	.CreateAccountView__linkContainer {
		text-align: center;
	}
}
@media only screen and (max-width: 500px)  {
	.CreateAccountView__headerTitle h3 {
		font-size: 20px;
		margin: 0 10px;
	}
}
