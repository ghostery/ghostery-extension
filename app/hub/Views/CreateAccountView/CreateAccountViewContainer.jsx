/**
 * Create Account View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
	validateEmail,
	validatePassword,
	validateEmailsMatch,
	validateConfirmEmail
} from '../../../panel/utils/utils';
import CreateAccountView from './CreateAccountView';
import SignedInView from '../SignedInView';

/**
 * @class Implement the Create Account View for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class CreateAccountViewContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			email: '',
			emailError: false,
			confirmEmail: '',
			confirmEmailError: false,
			firstName: '',
			lastName: '',
			password: '',
			passwordInvalidError: false,
			passwordLengthError: false,
			promotionsChecked: true,
			validateInput: false,
		};

		this.props.actions.setToast({
			toastMessage: '',
			toastClass: '',
		});
	}

	/**
	 * Update input values by updating state.
	 * @param  {Object} event the 'change' event
	 */
	_handleInputChange = (event) => {
		const { name, value } = event.target;
		this.setState({ [name]: value });

		if (!this.state.validateInput) {
			return;
		}

		switch (name) {
			case 'email': {
				const emailIsValid = value && validateEmail(value);
				this.setState({
					emailError: !emailIsValid,
				});
				break;
			}
			case 'confirmEmail': {
				const { email } = this.state;
				const confirmIsValid = value && validateEmailsMatch(email, value);
				this.setState({
					confirmEmailError: !confirmIsValid,
				});
				break;
			}
			case 'password': {
				const passwordIsValid = value && validatePassword(value);
				const invalidChars = !passwordIsValid && value.length >= 8 && value.length <= 50;
				const invalidLength = !passwordIsValid && !invalidChars;
				this.setState({
					passwordInvalidError: invalidChars,
					passwordLengthError: invalidLength,
				});
				break;
			}
			default: break;
		}
	}

	/**
	 * Update input checkbox values by updating state.
	 */
	_handleCheckboxChange = () => {
		const promotionsChecked = !this.state.promotionsChecked;
		this.setState({ promotionsChecked });
	}

	/**
	 * Handle creating an account, but validate the data first.
	 * @param  {Object} event the 'submit' event
	 */
	_handleCreateAccountAttempt = (event) => {
		event.preventDefault();
		const {
			email,
			confirmEmail,
			firstName,
			lastName,
			password,
			promotionsChecked
		} = this.state;
		const emailIsValid = email && validateEmail(email);
		const confirmIsValid = confirmEmail && validateConfirmEmail(email, confirmEmail);
		const passwordIsValid = password && validatePassword(password);
		const invalidChars = !passwordIsValid && password.length >= 8 && password.length <= 50;
		const invalidLength = !passwordIsValid && !invalidChars;

		this.setState({
			emailError: !emailIsValid,
			confirmEmailError: !confirmIsValid,
			passwordInvalidError: invalidChars,
			passwordLengthError: invalidLength,
			validateInput: true,
		});

		if (!emailIsValid || !confirmIsValid || !passwordIsValid) {
			return;
		}
		this.props.actions.setToast({
			toastMessage: '',
			toastClass: ''
		});
		this.props.actions.register(email, confirmEmail, firstName, lastName, password).then((success) => {
			if (success) {
				this.props.actions.updateAccountPromotions(promotionsChecked);
				this.props.actions.getUser();
				this.props.actions.setToast({
					toastMessage: t('hub_create_account_toast_success'),
					toastClass: 'success'
				});
				this.props.history.push('/');
			} else {
				this.props.actions.setToast({
					toastMessage: t('hub_create_account_toast_error'),
					toastClass: 'alert'
				});
			}
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Create Account View of the Hub app
	 */
	render() {
		const { loggedIn, user } = this.props;
		const {
			email,
			emailError,
			confirmEmail,
			confirmEmailError,
			firstName,
			lastName,
			password,
			passwordInvalidError,
			passwordLengthError,
			promotionsChecked,
		} = this.state;
		const createAccountChildProps = {
			email,
			emailError,
			confirmEmail,
			confirmEmailError,
			firstName,
			lastName,
			password,
			passwordInvalidError,
			passwordLengthError,
			promotionsChecked,
			handleInputChange: this._handleInputChange,
			handleCheckboxChange: this._handleCheckboxChange,
			handleSubmit: this._handleCreateAccountAttempt
		};
		const signedInChildProps = {
			email: user && user.email || email,
		};

		return loggedIn ? (
			<SignedInView {...signedInChildProps} />
		) : (
			<CreateAccountView {...createAccountChildProps} />
		);
	}
}

// PropTypes ensure we pass required props of the correct type
CreateAccountViewContainer.propTypes = {
	actions: PropTypes.shape({
		setToast: PropTypes.func.isRequired,
		register: PropTypes.func.isRequired,
		getUser: PropTypes.func.isRequired,
		updateAccountPromotions: PropTypes.func.isRequired,
	}).isRequired,
};

export default CreateAccountViewContainer;
