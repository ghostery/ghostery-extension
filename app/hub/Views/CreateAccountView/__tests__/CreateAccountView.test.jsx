/**
 * Create Account View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import CreateAccountView from '../CreateAccountView';

describe('app/hub/Views/CreateAccount component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('create account view is rendered correctly', () => {
			const initialState = {
				email: 'test@example.com',
				emailError: false,
				confirmEmail: 'test@example.com',
				confirmEmailError: false,
				firstName: 'First',
				lastName: 'Last',
				legalConsentChecked: true,
				password: '',
				passwordInvalidError: false,
				passwordLengthError: false,
				promotionsChecked: false,
				handleInputChange: () => {},
				handleLegalConsentCheckboxChange: () => {},
				handlePromotionsCheckboxChange: () => {},
				handleSubmit: () => {},
			};

			const component = renderer.create(
				<MemoryRouter>
					<CreateAccountView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('create account view is rendered correctly with empty inputs and errors', () => {
			const initialState = {
				email: '',
				emailError: true,
				confirmEmail: '',
				confirmEmailError: true,
				firstName: '',
				lastName: '',
				legalConsentChecked: false,
				password: '',
				passwordInvalidError: true,
				passwordLengthError: true,
				promotionsChecked: true,
				handleInputChange: () => {},
				handleLegalConsentCheckboxChange: () => {},
				handlePromotionsCheckboxChange: () => {},
				handleSubmit: () => {},
			};

			const component = renderer.create(
				<MemoryRouter>
					<CreateAccountView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				email: 'test@example.com',
				emailError: false,
				confirmEmail: 'test@example.com',
				confirmEmailError: false,
				firstName: 'First',
				lastName: 'Last',
				legalConsentChecked: true,
				password: '',
				passwordInvalidError: false,
				passwordLengthError: false,
				promotionsChecked: false,
				handleInputChange: () => {},
				handleLegalConsentCheckboxChange: () => {},
				handlePromotionsCheckboxChange: () => {},
				handleSubmit: jest.fn(),
			};

			const component = shallow(<CreateAccountView {...initialState} />);
			expect(component.find('.CreateAccountView').length).toBe(1);
			expect(component.find('.CreateAccountView__headerImage').length).toBe(1);
			expect(component.find('.CreateAccountView__inputBox').length).toBe(5);
			expect(component.find('button[type="submit"]').length).toBe(1);

			expect(component.find('.CreateAccountView__inputError').length).toBe(0);
			component.setProps({ emailError: true });
			expect(component.find('.CreateAccountView__inputError').length).toBe(1);

			expect(initialState.handleSubmit.mock.calls.length).toBe(0);
			component.find('form').simulate('submit');
			expect(initialState.handleSubmit.mock.calls.length).toBe(1);
		});
	});
});
