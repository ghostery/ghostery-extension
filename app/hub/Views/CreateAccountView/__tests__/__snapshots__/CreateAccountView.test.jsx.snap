// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/CreateAccount component Snapshot tests with react-test-renderer create account view is rendered correctly 1`] = `
<div
  className="CreateAccountView"
>
  <div
    className="CreateAccountView--addPaddingTop row align-center"
  >
    <div
      className="columns small-10"
    >
      <div
        className="CreateAccountView__header flex-container align-center-middle"
      >
        <img
          className="CreateAccountView__headerImage"
          src="/app/images/hub/account/ghosty-account.svg"
        />
        <div
          className="CreateAccountView__headerTitle"
        >
          <h3>
            hub_create_account_header_title
          </h3>
        </div>
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="CreateAccountView--addPaddingTop row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="email"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value="test@example.com"
        />
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email_confirm
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="confirmEmail"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      className="row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_first_name
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="firstName"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value="First"
        />
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email_last_name
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="lastName"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value="Last"
        />
      </div>
    </div>
    <div
      className="row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_password
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="password"
          onChange={[Function]}
          pattern=".{1,}"
          type="password"
          value=""
        />
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <div
          className="CreateAccountView__checkboxContainer CreateAccountView--marginBottom flex-container align-middle"
        >
          <div
            className="ToggleCheckbox ToggleCheckbox--active"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </div>
          <span
            className="CreateAccountView__inputLabel clickable"
            onClick={[Function]}
          >
            hub_create_account_checkbox_promotions
          </span>
        </div>
      </div>
    </div>
    <div
      className="row align-center"
    >
      <div
        className="CreateAccountView__linkContainer columns small-12 medium-10"
      >
        <a
          aria-current={null}
          className="CreateAccountView__link"
          href="/log-in"
          onClick={[Function]}
        >
          hub_create_account_link_login
        </a>
      </div>
    </div>
    <div
      className="CreateAccountView--addPaddingTop row align-center"
    >
      <div
        className="CreateAccountView__submit columns small-12 medium-10 flex-container flex-dir-row-reverse"
      >
        <button
          className="CreateAccountView__button button success"
          type="submit"
        >
          hub_create_account_button_submit
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`app/hub/Views/CreateAccount component Snapshot tests with react-test-renderer create account view is rendered correctly with empty inputs and errors 1`] = `
<div
  className="CreateAccountView"
>
  <div
    className="CreateAccountView--addPaddingTop row align-center"
  >
    <div
      className="columns small-10"
    >
      <div
        className="CreateAccountView__header flex-container align-center-middle"
      >
        <img
          className="CreateAccountView__headerImage"
          src="/app/images/hub/account/ghosty-account.svg"
        />
        <div
          className="CreateAccountView__headerTitle"
        >
          <h3>
            hub_create_account_header_title
          </h3>
        </div>
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="CreateAccountView--addPaddingTop row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox error"
          name="email"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value=""
        />
        <div
          className="CreateAccountView__inputError"
        >
          hub_create_account_label_email_invalid
        </div>
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email_confirm
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox error"
          name="confirmEmail"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value=""
        />
        <div
          className="CreateAccountView__inputError"
        >
          hub_create_account_label_email_confirm_invalid
        </div>
      </div>
    </div>
    <div
      className="row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_first_name
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="firstName"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value=""
        />
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_email_last_name
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox"
          name="lastName"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row align-center-middle"
    >
      <div
        className="columns small-12 medium-5"
      >
        <label
          className="CreateAccountView__inputLabel"
        >
          hub_create_account_label_password
        </label>
        <input
          autoComplete="off"
          className="CreateAccountView__inputBox error"
          name="password"
          onChange={[Function]}
          pattern=".{1,}"
          type="password"
          value=""
        />
        <div
          className="CreateAccountView__inputError"
        >
          hub_create_account_label_password_invalid
        </div>
        <div
          className="CreateAccountView__inputError"
        >
          hub_create_account_label_password_invalid_length
        </div>
      </div>
      <div
        className="columns small-12 medium-5"
      >
        <div
          className="CreateAccountView__checkboxContainer CreateAccountView--marginBottom flex-container align-middle"
        >
          <div
            className="ToggleCheckbox"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </div>
          <span
            className="CreateAccountView__inputLabel clickable"
            onClick={[Function]}
          >
            hub_create_account_checkbox_promotions
          </span>
        </div>
      </div>
    </div>
    <div
      className="row align-center"
    >
      <div
        className="CreateAccountView__linkContainer columns small-12 medium-10"
      >
        <a
          aria-current={null}
          className="CreateAccountView__link"
          href="/log-in"
          onClick={[Function]}
        >
          hub_create_account_link_login
        </a>
      </div>
    </div>
    <div
      className="CreateAccountView--addPaddingTop row align-center"
    >
      <div
        className="CreateAccountView__submit columns small-12 medium-10 flex-container flex-dir-row-reverse"
      >
        <button
          className="CreateAccountView__button button success"
          type="submit"
        >
          hub_create_account_button_submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
